backend/src/transactions/transactions.service.ts

```diff
@@ -1,28 +1,60 @@
-import { Injectable } from '@nestjs/common'
-import { CreateTransactionDto } from './dto/transaction.dto'
-import { Connection, Repository } from 'typeorm'
-import { InjectRepository } from '@nestjs/typeorm'
-import { Transactions } from './transaction.entity'
-import { User } from '../users/user.entity'
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { User } from '../users/user.entity';
+import { CreateTransactionDto } from './dto/transaction.dto';
+import { Transaction } from './transaction.entity';
+import { TransactionRepository } from './transaction.repository';
 
 @Injectable()
 export class TransactionsService {
   constructor(
-    @InjectRepository(Transactions)
-    private transactionRepository: Repository<Transactions>,
-  ) {
+    @InjectRepository(TransactionRepository) private transactionRepository: TransactionRepository) {
+  }
+
+  async getTransactions(
+    user: User
+  ): Promise<Transaction[]> {
+    return this.transactionRepository.getTransactions(user);
+  }
+
+  async getTransactionById(
+    id: number,
+    user: User
+  ): Promise<Transaction> {
+    const foundTransaction = await this.transactionRepository.findOne({ where: { id, userId: user.id } });
+
+    if (!foundTransaction) {
+      throw new NotFoundException(`Transaction with id ${id} was not found`);
+    }
+
+    return foundTransaction;
+  }
+
+  async createTransaction(createTransactionDto: CreateTransactionDto, user: User): Promise<Transaction> {
+    return this.transactionRepository.createTransaction(createTransactionDto, user);
+  }
+
+  async deleteTransaction(id: number, user: User): Promise<void> {
+    const transaction = await this.transactionRepository.delete({ id, userId: user.id });
+
+    if (transaction.affected === 0) {
+      throw new NotFoundException();
+    }
   }
 
-  // async create(createTransactionDto: CreateTransactionDto) {
-  //   const transaction = new Transactions()
-  //   transaction.amount = createTransactionDto.amount
-  //   transaction.description = createTransactionDto.description
-  //   transaction.date = createTransactionDto.date
-  //   transaction.categories = createTransactionDto.categories
-  //
-  //   const user = new User()
-  //   user.transactions = [transaction]
-  //
-  //   await this.transactionRepository.save(transaction)
-  // }
+  async updateTransaction(id: number, createTransactionDto: CreateTransactionDto, user: User): Promise<Transaction> {
+    const { amount, description, date, categories } = createTransactionDto;
+
+    const transaction = await this.getTransactionById(id, user);
+
+    transaction.amount = amount;
+    transaction.description = description;
+    transaction.date = date;
+    transaction.categories = categories;
+    transaction.user = user;
+    await transaction.save();
```

Comment:
Please use `this.transactionRepository.save(transaction)`

Reply:
Updated in separate PR. Thanks for the feedback.